staff/marti-malek/e-commerce-protos/scripts/cart.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * Demo of an e-commerce.
+ * 
+ * @param {Array} cartList 
+ */
+
+function Cart(cartList) {
+    this.cartList = []; 
+};
+
+/**
+ * Function that adds a product to the cart.
+ * 
+ * @throws {Error} -If too many arguments
+ */
+Cart.prototype.add = function (product) {
+    if (arguments.length > 1) throw Error ('too many arguments');
+
+    this.cartList.push(product);
+};
+
+/**
+ * Function that calculates the summed prices of all the products in the cart.
+ * 
+ * @returns {number} 
+ * 
+ * @throws {Error} - If too many arguments
+ */
+Cart.prototype.totalPrice = function () {
+    return this.cartList.reduce(function (accum, product){
+        return accum + product.price;
+    }, 0);
+    /* if (arguments.length > 0) throw Error ('too many arguments');
+
+    price2 = 0;
+    for(var i = 0; i < this.cartList.length; i++) {
+        price2 += this.cartList[i].price;
+    }
+    return price2; */
+};
+
+/**
+ * Function that calculates the total number of items added to the cart.
+ * 
+ * @returns {number} 
+ * 
+ * @throws {Error} - If too many arguments
+ */
+Cart.prototype.numberOfItems = function () {
+    if (arguments.length > 0) throw Error ('too many arguments');
+
+    var items = this.cartList.length;
+
+    return items;
+};
+
+/**
+ * Function that calculates the most expensive product of the ones added to the cart.
+ * 
+ * @returns {Object} 
+ * 
+ * @throws {Error} - If too many arguments
+ */
+Cart.prototype.mostExpensive = function () {
+    if (arguments.length > 0) throw Error ('too many arguments');
+
```

Comment:
Good job implementing the methods with reduce

Reply:
The merits go to @manuelbarzi. ;)